src/customFileParser/unpackGeometry/SphericalSegment.ts

```diff
@@ -0,0 +1,14 @@
+import { RenderedGeometryGroups } from '../sharedFileParserTypes';
+import PropertyLoader from '../PropertyLoader';
+
+function addOpenSphericalSegment(groups: RenderedGeometryGroups, data: PropertyLoader) {
+  groups.SphericalSegment.add(data.nodeId, data.treeIndex, data.color, data.center, data.normal,
+    data.radiusA, data.height);
+}
+
+function addClosedSphericalSegment(groups: RenderedGeometryGroups, data: PropertyLoader) {
+  addOpenSphericalSegment(groups, data);
+  groups.Circle.add(data.nodeId, data.treeIndex, data.color, data.center, data.normal, data.radiusA);
```

Comment:
Note that on ellipsoid segments and spherical segments, the radius of the circle isn't just the radius from the data.
For instance, the spherical segment data can be calculated as:
```
const { verticalRadius, horizontalRadius, height } = primitiveInfo;
const length = verticalRadius - height;
const circleRadius =
      (Math.sqrt(verticalRadius * verticalRadius - length * length) * horizontalRadius)
      / verticalRadius;
center.add(vector.copy(normal).normalize().multiplyScalar(length));
group.add(nodeId, treeIndex, color, center, normal, circleRadius);
```

Reply:
If you don't, you'll see the circles being placed at the wrong place :D